### DISCUSSION TITLE
How to add certificates to microcks containers

### DISCUSSION THREAD

**Discussion by Seb0042** [microcks/microcks#893] [Show and tell]
Hi,

We have some internals CA, so the "challenge" was to add them to Microcks containers.

You can use tool like keytool to add your own CA certificates to the standard one, and then mount this new keystore in the container.
If you're running on Kubernetes, the keystore can be stored as a binaryData in a configMap, maxsize of 1Mb.

Then the solution is based on JAVA_OPTIONS parameter. 
For example, you can have this:
```
JAVA_OPTIONS: "-Djavax.net.ssl.trustStore=/tmp/.keystore -Djavax.net.ssl.trustStorePassword=changeit"
```

**Comment by lbroudoux** [comment]
Thanks for the explanations! As we do not support (yet!) mounting extra-volumes on Microcks pods in Kubernetes, I supposed you run this configuration through docker-compose or podman-compose. So it also means you had to add an extra volume mount in the compose file? Am I correct?

**Reply by Seb0042** [reply]
Yes, podman-compose, we add a new volume in the compose file.
For kubernetes you can add a configmap and modify the deployment may be ? (with helm, with the operator it may be less easy).
